node_modules/oae-principals/tests/test-guests.js

```diff
@@ -0,0 +1,437 @@
+        globalAdminRestContext = TestsUtil.createGlobalAdminRestContext();
+
+        // Make sure there are no pending emails in the email buckets
+        EmailTestUtil.clearEmailCollections(function() {
+
+            // Send out daily emails
+            ActivityTestUtil.refreshMailConfiguration(null, true, false, {}, function(config) {
+                return callback();
+            });
+        });
+    });
+
+    /**
+     * Function that will disable the daily email collector
+     */
+    after(function(callback) {
+        // Disable daily emails
+        ActivityTestUtil.refreshMailConfiguration(null, false, false, {}, function(config) {
+            return callback();
+        });
+    });
+
+    /**
+     * Test that verifies that guests can be invited to collaborate on a resource
+     */
+    it('verify that guests can be invited to collaborate on a single resource', function(callback) {
+        ContentTestUtil.createContentAndInviteGuest(camAdminRestContext, function(simong, link1, email, token) {
+            return callback();
+        });
+    });
+
+    /**
+     * Test that verifies that guests can be invited to collaborate on multiple resources
+     */
+    it('verify that guests can be invited to collaborate on multiple resources', function(callback) {
+        // Create some content and invite a guest
+        ContentTestUtil.createContentAndInviteGuest(camAdminRestContext, function(simong, link1, email, token) {
+
+            // Create another content item and invite the guest again
+            RestAPI.Content.createLink(simong.restContext, 'Apereo', 'Apereo', 'public', 'http://www.apereo.org', [], [], [], function(err, link2) {
+                assert.ok(!err);
+                RestAPI.Content.shareContent(simong.restContext, link2.id, [email], function(err) {
+                    assert.ok(!err);
+
+                    // Create a guest account with the token that was emailed to the guest
+                    PrincipalsTestUtil.signIntoGuestAccountWithToken(token, function(loggedinGuest) {
+
+                        // Verify both the content items are in the guest's content library
+                        RestAPI.Content.getLibrary(loggedinGuest.restContext, loggedinGuest.user.id, null, null, function(err, data) {
+                            assert.ok(!err);
+                            assert.strictEqual(data.results.length, 2);
+                            assert.strictEqual(data.results[0].id, link2.id);
+                            assert.strictEqual(data.results[1].id, link1.id);
+                            return callback();
+                        });
+                    });
+                });
+            });
+        });
+    });
+
+    /**
+     * Test that verifies that guests are created on the guests tenant if no tenant email domain matches
+     */
+    it('verify that guests are created on the guests tenant if no tenant email domain matches', function(callback) {
```

Comment:
We should also verify that this does not work when the guest is being invited from a private tenant